File: Huan_link_all_script/workspace/drugrepo/dataset/gene-disease/Cancer/somatic_mutation/ICGC/before/refer/step1.pl
#!/usr/bin/perl
use warnings;
use strict;

# my $fi_tags ="./somatic.vcf.bed";
# open my $fh_tags, '<', $fi_tags or die "$0 : failed to open input file '$fi_tags' : $!\n";

open my $DATE, 'zless somatic.vcf.bed|' or die "zless somatic.vcf.bed $0: $!\n"; 

while(<$DATE>)
{

     chomp();  
      my @f1 = split /\s+/;
      my $chrom = $f1[0];
      my $pos = $f1[2];
      my $ID = $f1[3];
      my $ref = $f1[5];
      my $alt = $f1[6];
   
           
           #print $chrom."_".$pos."\t"..$ref."\t".$alt."\n"
           print $chrom." ".$pos." \. ".$ref." ".$alt." \. \. \."."\n";
         






}

File: Huan_link_all_script/workspace/drugrepo/dataset/gene-disease/Cancer/somatic_mutation/ICGC/before/refer/step1.py
import pandas as pd
import numpy as np
import vcf
import sys

#chr_id = sys.argv[1]
vcf_reader=vcf.Reader(open('./simple_somatic_mutation.aggregated' + '.vcf.gz','r'))

out_file  = open('./simple_somatic_mutation.aggregated' + '.txt','w')

for record in vcf_reader:
    tmp = record.INFO['OCCURRENCE']
    #print type(tmp[0])
    value_tmp = tmp[0].split('|')
    #print type(float(value_tmp[3]))
    ID = str (record.ID)
    print type(ID)
    break
    # CHROM=str(record.CHROM)
    # POS=str(record.POS)
    # ID = str (record.ID)
    # REF=str(record.REF)
    # ALT=str(record.ALT)
    # INFO = str(record.INFO)
    # out_str = CHROM +'\t' + POS + '\t' + REF + '\t' + ALT + '\t' + ID + '\t' +INFO + '\n'
    out_file.write(out_str)
out_file.close()